lib/http.go

```diff
@@ -0,0 +1,50 @@
+package lib
+
+// Based on https://gist.github.com/josue/598ac048f78362f593f49436680b60b7
+
+import (
+	"log"
+	"net/http"
+	"strings"
+)
+
+/*
+GetURLHeaders returns a map array of all available headers.
+@param string - URL given
+@return map[string]string
+*/
+func GetURLHeaders(url string) map[string]string {
+	response, err := http.Head(url)
+	if err != nil {
+		log.Fatal("Error: Unable to download URL (", url, ") with error: ", err)
+	}
+
+	if response.StatusCode != http.StatusOK {
+		log.Fatal("Error: HTTP Status = ", response.Status)
+	}
+
+	headers := make(map[string]string, len(response.Header))
+
+	for k, v := range response.Header {
```

Comment:
There's some header name canonicalization built into `response.Header` when you use its `Get()` and `Set()` methods, I believe.